### PR TITLE
[FIX] web: o2m correctly left-align the first create button

### PR DESCRIPTION
**Pull request by seb-odoo**
When there is no handle field, there was an unnecessary margin-left. When there was a handle field, the margin-left wasn't the correct way to align it.

PR: #
Task: 1869469

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/list/list_editable_renderer.js [hunk 1]
```diff
@@ -572,17 +575,27 @@ ListRenderer.include({
         var $rows = this._super();
 
         if (this.addCreateLine) {
+            var $tr = $('<tr>');
+            var numberOfCols = self._getNumberOfCols();
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_editable_renderer.js]: i'd name it colspan, for sake of clarity

File: addons/web/static/tests/fields/relational_fields_tests.js [hunk 2]
```diff
@@ -9862,8 +9862,10 @@ QUnit.module('relational_fields', {
         });
 
         // new controls correctly added
-        assert.strictEqual($('.o_field_x2many_list_row_add').length, 1);
-        assert.strictEqual($('.o_field_x2many_list_row_add').text(), "Add foodAdd pizzaAdd pasta");
+        var $td = form.$('.o_field_x2many_list_row_add');
+        assert.strictEqual($td.length, 1);
+        assert.strictEqual($td.parent().find('td').length, 1);
```
Comment (aab-odoo) [inline addons/web/static/tests/fields/relational_fields_tests.js]: Not a fan of this selector

Comment (seb-odoo) [reply addons/web/static/tests/fields/relational_fields_tests.js]: closest with a selector (tr) or better

**Comment by seb-odoo** [comment]
@aab-odoo I updated the first commit with your recommandations

**Comment by seb-odoo** [comment]
@ged-odoo here is the PR with a first commit (already reviewed by aab) and the new commit of today concerning the section widgets that I would like you to review. Thanks!

**Pull request merged**